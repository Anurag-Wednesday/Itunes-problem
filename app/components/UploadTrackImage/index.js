import React from 'react';
import { Button, Form, Upload } from 'antd';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { UploadOutlined } from '@ant-design/icons';
import * as colors from '@app/themes/colors';
import T from '@components/T';

const StyledFormItem = styled(Form.Item)`
  && {
    margin: 2rem;
    color: ${colors.whiteText};
    wrapper
  }
  .ant-form-item-required {
    color: ${colors.whiteText};
    font-size: 1rem;
  }
`;

const UploadTrackImage = ({ uploadProps }) => {
  return (
    <Form
      name="basic"
      labelCol={{ span: 8 }}
      wrapperCol={{ span: 16 }}
      initialValues={{ remember: false }}
      autoComplete="on"
    >
      <StyledFormItem
        label={<T id="input-artwork" />}
        name="artWorkUrl100"
        rules={[{ required: true, message: <T id="input-artwork" /> }]}
      >
        <Upload {...uploadProps}>
          <Button id="upload" data-testid="upload" icon={<UploadOutlined />}>
            <T id="upload_text" />
          </Button>
        </Upload>
      </StyledFormItem>
      <StyledFormItem wrapperCol={{ offset: 8, span: 16 }}>
        <Button type="primary" htmlType="submit" data-testid="submit">
          <T id="submit" />
        </Button>
      </StyledFormItem>
    </Form>
  );
};

UploadTrackImage.propTypes = {
  uploadProps: PropTypes.object
};
export default UploadTrackImage;
