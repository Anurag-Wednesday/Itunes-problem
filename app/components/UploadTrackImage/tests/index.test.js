import React from 'react';
import { fireEvent } from '@testing-library/dom';
import { renderWithIntl, timeout } from '@utils/testUtils';
import UploadTrackImage from '../index';

describe('<UploadTrack />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<UploadTrackImage />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 UploadTrack component', () => {
    const { getAllByTestId } = renderWithIntl(<UploadTrackImage />);
    expect(getAllByTestId('upload').length).toBe(1);
  });

  it('should give user the option to upload Image', async () => {
    const handleOnSubmit = jest.fn();
    const { getByTestId } = renderWithIntl(<UploadTrackImage handleOnSubmit={handleOnSubmit} />);
    const file = new File([''], 'chucknorris.png', { type: 'image/png' });
    fireEvent.change(getByTestId('upload'), {
      target: { files: [file] }
    });
    await timeout(500);
    let image = document.getElementById('upload');
    expect(image.files[0].name).toBe('chucknorris.png');
    expect(image.files.length).toBe(1);
    fireEvent.click(getByTestId('submit'));
  });

  it('should upload the image', () => {
    const uploadProps = {
      action: 'https://www.mocky.io/v2/5cc8019d300000980a055e76',
      defaultFileList: [
        {
          uid: '1',
          name: 'xxx.png',
          status: 'done',
          response: 'Server Error 500',
          url: 'http://www.baidu.com/xxx.png'
        }
      ]
    };
    const { getByTestId, baseElement } = renderWithIntl(<UploadTrackImage uploadProps={uploadProps} />);
    fireEvent.click(getByTestId('submit'));
    expect(baseElement.getElementsByClassName('ant-upload-list-item').length).toBe(1);
  });
});
