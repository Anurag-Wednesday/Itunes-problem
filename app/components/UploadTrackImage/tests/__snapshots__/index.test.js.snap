// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UploadTrack /> should render and match the snapshot 1`] = `
<body>
  <div>
    <form
      autocomplete="on"
      class="ant-form ant-form-horizontal"
      id="basic"
    >
      <div
        class="ant-row ant-form-item UploadTrackImage__StyledFormItem-krfdq7-0 dgYrEo"
      >
        <div
          class="ant-col ant-col-8 ant-form-item-label"
        >
          <label
            class="ant-form-item-required"
            for="basic_artWorkUrl100"
            title=""
          >
            <p
              class="T__StyledText-gjlic1-0 iVJYgn"
              data-testid="t"
            >
              Please input artwork
            </p>
          </label>
        </div>
        <div
          class="ant-col ant-col-16 ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <span
                class=""
              >
                <div
                  class="ant-upload ant-upload-select ant-upload-select-text"
                >
                  <span
                    class="ant-upload"
                    role="button"
                    tabindex="0"
                  >
                    <input
                      accept=""
                      id="basic_artWorkUrl100"
                      style="display: none;"
                      type="file"
                    />
                    <button
                      class="ant-btn"
                      data-testid="upload"
                      id="upload"
                      type="button"
                    >
                      <span
                        aria-label="upload"
                        class="anticon anticon-upload"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="upload"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M400 317.7h73.9V656c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V317.7H624c6.7 0 10.4-7.7 6.3-12.9L518.3 163a8 8 0 00-12.6 0l-112 141.7c-4.1 5.3-.4 13 6.3 13zM878 626h-60c-4.4 0-8 3.6-8 8v154H214V634c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v198c0 17.7 14.3 32 32 32h684c17.7 0 32-14.3 32-32V634c0-4.4-3.6-8-8-8z"
                          />
                        </svg>
                      </span>
                      <p
                        class="T__StyledText-gjlic1-0 iVJYgn"
                        data-testid="t"
                      >
                        Click to upload
                      </p>
                    </button>
                  </span>
                </div>
                <div
                  class="ant-upload-list ant-upload-list-text"
                />
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-row ant-form-item UploadTrackImage__StyledFormItem-krfdq7-0 dgYrEo"
      >
        <div
          class="ant-col ant-col-16 ant-col-offset-8 ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <button
                class="ant-btn ant-btn-primary"
                data-testid="submit"
                type="submit"
              >
                <p
                  class="T__StyledText-gjlic1-0 iVJYgn"
                  data-testid="t"
                >
                  Submit
                </p>
              </button>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</body>
`;
